lib/media/eme_manager.js

```diff
@@ -705,6 +706,63 @@ shaka.media.EmeManager.prototype.requestLicense_ = function(
 
 
 /**
+ * Standard pre-processor for PlayReady license requests.
+ *
+ * @param {!shaka.player.DrmInfo.LicenseRequestInfo} info License request info.
+ *
+ * @private
+ */
+shaka.media.EmeManager.prototype.playReadyLicensePreProcessor_ =
+    function(info) {
+  /*
+  The playready license body is actually an XML string, so need to convert
+  info.body (which is a Uint8Array, holding UTF-16 text data) to a string
+
+  XML typically has this structure (as an example):
+  <PlayReadyKeyMessage type="LicenseAcquisition">
+    <LicenseAcquisition Version="1">
+      <Challenge encoding="base64encoded">
+        {Base64EncodedBinaryChallengeData}
+      </Challenge>
+      <HttpHeaders>
+        <HttpHeader>
+          <name>Content-Type</name>
+          <value>text/xml; charset=utf-8</value>
+        </HttpHeader>
+        <HttpHeader>
+          <name>SOAPAction</name>
+          <value>
+            "http://schemas.microsoft.com/DRM/2007/03/protocols/AcquireLicense"
+          </value>
+        </HttpHeader>
+      </HttpHeaders>
+    </LicenseAcquisition>
+  </PlayReadyKeyMessage>"
+
+  Only challenge data is sent to the server, http errors are required to be
```

Comment:
Whoops, yup I meant http headers. Will reword that whole comment, it's a bit clunky...